artscraper/find_artworks.py

```diff
@@ -209,26 +209,77 @@ def get_artist_works(self):
         # Find the parent element corresponding to the text heading
         parent_element = element.find_element('xpath', '../..')
 
-        # Find right arrow button
-        right_arrow_element = parent_element.find_element('xpath', \
-            './/*[contains(@data-gaaction,"rightArrow")]')
+        # Find number of artists
+        # Find elements with tag name 'h3'
+        items_elements = parent_element.find_elements('tag name', 'h3')
+        for element in items_elements:
+            if 'items' in element.text:
+                match = re.search(r'\d+', element.text)
+                if match:
+                    total_num_artworks = int(match.group())
+                    break
+
+        # Initialize number of artworks
+        num_artworks = 0
+        # Initialize count of number of iterations for which the number of artworks remains the same
+        count = 0
+
+        while True:
```

Comment:
I think there we're still in the possible danger zone of infinite loops. If the right arrow button can be clicked, but won't add new elements to the list, then it will continue to loop.

There also seems to be some duplication in this function. Below a pseudo-code suggestion to make this part of the code easier to read.

```python
art_list = _get_art_list(...)
while (len(art_list) < tot_art)) and not (tot_art == 0 and n_try > 3):
    if _button_clickable():
        wait()
        click_button()
    art_list = _get_art_list()
    if len(art_list) == prev_len:
       n_try += 1
    else:
       n_try = 0
```

It might also be helpful to put theses part of the code in small functions (you can define them inline). Please check if the logic above is correct if you decide to implement it that way!